### DISCUSSION TITLE
Update DXCC data

### DISCUSSION THREAD

**Discussion by jarrah791** [wavelog/wavelog#774] [Q&A] [answered]
Hi, my server (shared webhosting) only allows 180s PHP execution time. That's not enough to update DXCC data. How can I update this data in a different way?

**Comment by HB9HIL** [comment]
hello @jarrah791 

I tested the DXCC Update on my slowest machine. An untuned standard install on a raspberry pi 4. 

This took 55 seconds to finish. 

My questions to you:

1. Which version of Wavelog are you running?
2. Do you talk about the DXCC Update in the installer or in the admin menu? 

180 second should be more then enough to run the DXCC Update

**Reply by jarrah791** [reply]
Hi HB9HIL, thanks for your reply and for testing with your hardware.

1. I'm running the latest stable version 1.8.2. Installed it today by cloning from git. 
2. I've experienced this problem both in the installer and in the admin menu. In the installer menu the installation seemed to have failed exactly after 180s (ajax failed), but in fact the installation was correct except of the DXCC updates. I've installed a second time with the same result. The installation is working fine except these updates which are impossible in the admin menu as well. Even if I wait for more than 20 minutes all I can see is 
"Bereite DXCC-Ausnahmen vor: 21120
DXCC-Entitäten: 402
DXCC-Ausnahmen: 0
DXCC-Präfixe: 0"
When reloading the page I'm invited again to update DXCC data. I've tried several times, cleared browser cache ... no difference. 

During installation the install wizard marked that 180s might be not enough but everything else was marked with a green mark.

**Reply by HB9HIL** [reply]
This does not sound like a normal behaviour. You probably should enable error logging (set `log_threshold` to 1 or max level 2 in your config.php). Then try to run the DXCC Update again and see if you got any logs in `application/logs`

**Reply by jarrah791** [reply]
I've got a log. It says "Migration class already loaded. Second attempt ignored." for every time I try to start updating. 

During installation the debug log noted "Could not update DXCC data. Ajax crashed."

But on the Wavelog Debug page there's the last time of download noted without any problems visible.

Hope this helps.

**Reply by HB9HIL** [reply]
which Migration Version do you see in the Debug View?

<img width="643" alt="image" src="https://github.com/user-attachments/assets/9457161d-feee-4ba5-aa11-17cf2a7aff5e">

It should say 213.

Also please tell me the MySQL/MariaDB Version.

**Reply by jarrah791** [reply]
My migration version is 213. 
The MySQL version is 8.0.33

**Reply by HB9HIL** [reply]
And everything is green in debug? (except the 180s thing) ..

mh no Idea what the issue could be here. But shared hostings are always kind of a mysterybox

**Reply by jarrah791** [reply]
Yes, everything is green except of the max_execution_time.

**Reply by HB9HIL** [reply]
Does the SCP File update work?

**Reply by jarrah791** [reply]
The SCP update works without any problems :-) But the log says for SCP "Paths class already loaded. Second attempt ignored."

**Reply by HB9HIL** [reply]
Okay. So no issue with Clublog (One of the SCP Files also come from Clublog, same as the DXCC data). 

to be honest. I have no Clue whats wrong with the DXCC Update on your instance. You could check your browser console (Press F12 in Chrome) if you got any error's here. 

Those "Second attempt ignored" messages, are fully okay. Just ignore them

**Reply by jarrah791** [reply]
There's an error:
![Zwischenablage01](https://github.com/user-attachments/assets/eea3905e-2cdc-4b9b-b29d-7d37017aabcd)
![Zwischenablage02](https://github.com/user-attachments/assets/a528438e-b26d-4494-96df-21ddf72f7cd6)

**Reply by HB9HIL** [reply]
Do you have access to the webserver error log? Not the wavelog log, I mean the error.log of apache/nginx

**Reply by HB9HIL** [reply]
And please try to call this URL directly `https://[wavelog-url]/index.php/update/dxcc`. Make sure you use the browser where you also logged in in your wavelog instance

**Reply by jarrah791** [reply]
Yes, I do have access to the error.log of the webserver. It's fine, nothing in relation to this problem. 
I've tried the URL directly - with a 504 as result.

**Reply by HB9HIL** [reply]
To be honest. No clue whats wrong here. Could be everything between file permissions, shared hosting and the shared hosters database setup. If you want you can send me an email hb9hil@wavelog.org and I can take a look directly on your instance. But in fact it is nothing Wavelog related as on every testsystem here the dxcc update works without issues. 

The most likely problem is still that your webservers IP adress is blocked by Clublog. To check that call the URL

`https://[wavelog-url]/index.php/update/update_clublog_scp`

You should see a blank white page with the text

`DONE: 224,818 callsigns loadedDONE: 50,323 callsigns loaded`

If this is no issue we'll have to take a look deeper in your installation.

**Reply by jarrah791** [reply]
The URL has this result: DONE: 224,818 callsigns loadedDONE: 50,323 callsigns loaded :-) 

Thank you so much for your help until now and for the offer of helping me!! Before I'll send you an email I'll try to reinstall the whole system tomorrow. If there'll be still a problem I'll contact you again. Thanks for today & vy 73 de DL4AI

**Reply by jarrah791** [reply, marked as answer]
Hi! I just want to give you a feedback. Unfortunately there was no way with my previous shared hosting to get Wavelog to work properly. I've even asked to change time to 600s but the provider said no. So I've changed my provider and it works like a charm. That's not a solution for the problem but there wasn't any. Thank you very much for your support! Great work!

**Reply by HB9HIL** [reply]
Awesome feedback! Thank you very much :)

**Reply by K9bdh** [reply]
I am also having issue of doing dxcc country update.   I tried the above but it says it can't open page but rest of wave log appears to work?       when I switch it from https to http.  I get this.DONE: 228,231 callsigns loadedDONE: 50,060 callsigns loaded

what do I need to switch?
